Simplify Suppliers form

JevgenijVisockij opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplifying  supplier form rendering
| Type?         | refacto
| Category?     | BO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | Part of #16482
| How to test?  | Supplier form must look and work exactly as before. There is one exception is that description no longer has a hint about invalid characters as that hint was simply not true.

:notebook: **BC Break**
Backwards compatibility because of changed construct parameters for suppliers type. If any module extends that type they will get an exception upon updating to version containing this PR.
<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/19996)
<!-- Reviewable:end -->

LouiseBonnard commented on src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php:
```diff
@@ -150,26 +190,28 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ->add('phone', TextType::class, [
-                'empty_data' => '',
+                'label' => $this->trans('Phone', 'Admin.Global'),
+                'help' => $this->trans('Phone number for this supplier', 'Admin.Catalog.Help'),
```
```suggestion
                'help' => $this->trans('Phone number for this supplier.', 'Admin.Catalog.Help'),
```

LouiseBonnard replied:
In order to stay consistent with the other help texts. ;-)

LouiseBonnard commented on src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php:
```diff
@@ -150,26 +190,28 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ])
             ->add('post_code', TextType::class, [
-                'empty_data' => '',
+                'label' => $this->trans('Zip/postal code', 'Admin.Global'),
```
@Junebyun, I can see this string also exists with a capital letter (_Zip/Postal code_): which one do you recommend?

Junebyun replied:
I'll go with the capital! It would be consistent with the capital letter of Zip! Thanks. :)

LouiseBonnard replied:
It suits me fine! I let you add this decision to the wording guidelines [document](https://docs.google.com/document/d/1pI5fsmQO9Zp2PSDRicnFHuBGV1j2xpG4h5z9K4Bl4Bg/edit), thank you.

Junebyun commented on src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php:
```diff
@@ -150,26 +190,28 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ])
             ->add('post_code', TextType::class, [
-                'empty_data' => '',
+                'label' => $this->trans('Zip/postal code', 'Admin.Global'),
```
```suggestion
                'label' => $this->trans('Zip/Postal code', 'Admin.Global'),
```

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php

Adding dot to the help text

Co-authored-by: LouiseBonnard <32565890+LouiseBonnard@users.noreply.github.com>

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php

Changing Zip/postal code to capital

Co-authored-by: Junebyun <64894223+Junebyun@users.noreply.github.com>

JevgenijVisockij commented:
Still WIP because needs choosing state solved which is done in https://github.com/PrestaShop/PrestaShop/pull/20016/ will finish, once that is done.

davidglezz commented on src/Core/ConstraintValidator/TypedRegexValidator.php:
```diff
@@ -41,6 +41,8 @@
  */
 class TypedRegexValidator extends ConstraintValidator
 {
+    const GENERIC_NAME_CHARS = '<>={}';
```
`GENERIC_NAME_FORBIDDEN_CHARS` is more descriptive

davidglezz reviewed (changes requested):
Nice

JevgenijVisockij committed:
Simplify supplier form rendering

# Conflicts:
#	src/Core/ConstraintValidator/TypedRegexValidator.php
#	src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php
#	src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Suppliers/Blocks/form.html.twig

JevgenijVisockij committed:
Fixing conflicts

# Conflicts:
#	src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php

JevgenijVisockij commented:
This is WIP until https://github.com/PrestaShop/PrestaShop/pull/21115 is merged.

Julievrz commented on src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php:
```diff
@@ -150,26 +190,28 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ->add('mobile_phone', TextType::class, [
-                'empty_data' => '',
+                'label' => $this->trans('Mobile phone', 'Admin.Global'),
+                'help' => $this->trans('Mobile phone number for this supplier.', 'Admin.Catalog.Help'),
```
Hi! Regarding this wording, I can't see the help text on the software at the moment? Is it really necessary to have a help text there? Wtdy @LouiseBonnard 
![image](https://user-images.githubusercontent.com/70583503/101499814-fafc7680-396d-11eb-9756-06f80d4573e1.png)

Julievrz replied:
(I checked 1.7.7 version)

LouiseBonnard replied:
I agree, the two phone fields are explicit enough to me.

JevgenijVisockij replied:
Removed the help from phone and mobile phone.

JevgenijVisockij committed:
Removed help from phone and mobile phone

Julievrz commented on src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php:
```diff
@@ -103,17 +118,36 @@ public function __construct(
+
+        $invalidCharsText = sprintf(
+            '%s ' . TypedRegexValidator::GENERIC_NAME_CHARS,
+            $this->trans('Invalid characters:', 'Admin.Notifications.Info')
```
```suggestion
            $this->trans('Invalid characters:', 'Admin.Global')
```

Julievrz replied:
Sorry I missed this one but I corrected it in other PRs so I'll do it here too for more consistency. 👍

Julievrz commented on src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php:
```diff
@@ -103,17 +118,36 @@ public function __construct(
+
+        $invalidGenericNameHint = sprintf(
+            '%s <>={}',
+            $this->trans('Invalid characters:', 'Admin.Notifications.Info')
```
```suggestion
            $this->trans('Invalid characters:', 'Admin.Global')
```

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php

Changed domain

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php

Changed domain

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

JevgenijVisockij committed:
Added components to supplier form

Progi1984 commented on src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Suppliers/Blocks/form.html.twig:
```diff
@@ -34,77 +34,6 @@
-        'help': 'Upload a supplier logo from your computer.'|trans({}, 'Admin.Catalog.Help')
-      }) }}
-
       {% if logoImage is defined and logoImage is not null %}
```
Why do you change order for this element ?

JevgenijVisockij replied:
Yeah that was a mistake, moved it where it should be. Thanks!

Progi1984 reviewed (commented):
A question

JevgenijVisockij committed:
Fixed place of logo element

JevgenijVisockij committed:
Fixed characters hint

JevgenijVisockij committed:
Removed hint from suppliers description as it didn't make sense

atomiix commented:
@JevgenijVisockij I don't think this is a WIP anymore, right?

JevgenijVisockij commented:
> @JevgenijVisockij I don't think this is a WIP anymore, right?

You are right, it's not.

khouloudbelguith commented:
Hi @JevgenijVisockij,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1JCfgJC9j1Gx7kR4-br00ktoOmWt1uJrg/view?usp=sharing

Tnaks!

PululuK commented:
Thanks @JevgenijVisockij 
Don't forget to update https://github.com/PrestaShop/PrestaShop/issues/16482 ... I can do it if you want 🙂
